bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
+      exit 1
+    fi
+
+  fi
+
+}
+
+function lambda_deploy () {
+
+  # check environment variables
+  if [[ -z "${DEPLOY_AWS_ACCOUNT}" ]] ; then
+    echo "Error: DEPLOY_AWS_ACCOUNT is required"
+    exit 1
+  fi
+
+  if [[ -z "${DEPLOY_LAMBDA_FUNCTION_NAME}" ]] ; then
+    echo "Error: DEPLOY_LAMBDA_FUNCTION_NAME is required"
+    exit 1
+  fi
+
+  if [[ -z "${DEPLOY_LAMBDA_FUNCTION_ROLE}" ]] ; then
+    echo "Error: DEPLOY_LAMBDA_FUNCTION_ROLE is required"
+    exit 1
+  fi
+
+  # setup aws
+  aws_configure
+
+  # zip the
+  if [[ ! -z "${DEPLOY_LAMBDA_ZIP_FULLPATH}" ]] && [[ -f "${DEPLOY_LAMBDA_ZIP_FULLPATH}" ]] ; then
+    echo "Skipping ... Zip already exists at ${DEPLOY_LAMBDA_ZIP_FULLPATH}"
+  else
+    lambda_package_zip
+  fi
+
+  lamdba_runtime="${DEPLOY_LAMBDA_FUNCTION_RUNTIME:-nodejs6.10}"
+  lamdba_handler="${DEPLOY_LAMBDA_FUNCTION_HANDLER:-index.handler}"
+  lambda_role="arn:aws:iam::${DEPLOY_AWS_ACCOUNT}:role/${DEPLOY_LAMBDA_FUNCTION_ROLE}"
+  lambda_timeout="${DEPLOY_LAMBDA_FUNCTION_TIMEOUT:-3}"
+  lambda_memory_size="${DEPLOY_LAMBDA_FUNCTION_MEMORY_SIZE:-128}"
+
+  lambda_env_template="${DEPLOY_LAMBDA_FUNCTION_ENV_TEMPLATE:-environment.txt}"
+
+  lambda_env=$( \
+    ${DEPLOY_ENVSUBST_COMMAND:-envsubst} < "${lambda_env_template}"
+  )
+
+  # create lambda function
+  lambda_function_arn=$( \
+    aws lambda create-function \
+    --profile "${DEPLOY_AWS_PROFILE}" \
+    --function-name "${DEPLOY_LAMBDA_FUNCTION_NAME}" \
+    --description "${DEPLOY_LAMBDA_FUNCTION_DESCRIPTION}" \
+    --timeout "${lambda_timeout}" \
+    --memory-size "${lambda_memory_size}" \
+    --role "${lambda_role}" \
+    --handler "${lamdba_handler}" \
+    --runtime "${lamdba_runtime}" \
+    --environment "${lambda_env}" \
+    --zip-file "fileb://${DEPLOY_LAMBDA_ZIP_FULLPATH}" \
+    | $JQ ".FunctionArn"
+  )
+
+  # try to update the lamda function if creation failed
+  if [[ -z "$lambda_function_arn" ]] ; then
+
     lambda_function_arn=$( \
-        aws lambda create-function \
-            --profile "${DEPLOY_AWS_PROFILE}" \
-            --function-name "${DEPLOY_LAMBDA_FUNCTION_NAME}" \
-            --description "${DEPLOY_LAMBDA_FUNCTION_DESCRIPTION}" \
-            --timeout "${lambda_timeout}" \
-            --role "${lambda_role}" \
-            --handler "${lamdba_handler}" \
-            --runtime "${lamdba_runtime}" \
-            --environment "${lambda_env}" \
-            --zip-file "fileb://../${DEPLOY_LAMBDA_FUNCTION_NAME}.zip" \
-            | $JQ ".FunctionArn"
+      aws lambda update-function-configuration \
+      --profile "${DEPLOY_AWS_PROFILE}" \
+      --function-name "${DEPLOY_LAMBDA_FUNCTION_NAME}" \
+      --description "${DEPLOY_LAMBDA_FUNCTION_DESCRIPTION}" \
+      --timeout "${lambda_timeout}" \
+      --memory-size "${lambda_memory_size}" \
+      --role "${lambda_role}" \
+      --handler "${lamdba_handler}" \
+      --environment "${lambda_env}" \
+      --runtime "${lamdba_runtime}" \
+      | $JQ ".FunctionArn"
```

Comment:
Is there a way to consolidate this `update-function-configuration` call's flags and the `create-function` call's flags (above)? The only difference in the arguments that I see is that the `create-function` call passes in a `--zip-file ...`

Reply:
Hmm, good point, I'll see about that.

Reply:
Actually, more I read through this, I don't think it can be done. Basically this is only running if the previous command failed. Basically it's doing a very ghetto upsert.